### DISCUSSION TITLE
Use Handle as Source and Target

### DISCUSSION THREAD

**Discussion by ghost** [xyflow/xyflow#1537] [Q&A]
Hi,
I have a usecase, where a handle needs to be both source and target. is there any way to make this work, since HandleType only allows for source OR target?

**Comment by ArturMakly** [comment]
i have the exact same need ( related to: https://github.com/wbkd/react-flow/discussions/1331 )
any luck solving this? @konsti-papa

**Reply by moklick** [reply]
you can set `connectionMode="loose"` and then only use handle type `"source"`

**Reply by ghost** [reply]
@moklick but where can you set the connection mode? I'd need it loose for only one type of edge...

**Reply by datoslabs** [reply]
The property is in `<ReactFlow>`, see the [API doc](https://reactflow.dev/docs/api/component-props/)

**Reply by zhenghan2017** [reply]
@moklick this way is active, but if i have self-connect edge, it can't work, because sourcePositon and targetPosition will be same.how can resolve it?

**Reply by KuwadaKouhei** [reply]
You can solve this by implementing both the `type: source` handle and `type: target` handle overlapping.
